### PR TITLE
qf-tests: fix KNM tests

### PR DESCRIPTION
**Pull request by fenollp**

### CODE REVIEW

**Comment by fenollp** [comment]
Wait, working on implementing bulk.

File: core/kazoo_number_manager/src/carriers/knm_other.erl [hunk 1]
```diff
@@ -57,9 +57,9 @@ is_local() -> 'false'.
 %% in a rate center
 %% @end
 %%--------------------------------------------------------------------
--spec find_numbers(ne_binary(), pos_integer(), knm_carriers:options()) ->
-                          {'ok', knm_number:knm_numbers()} |
-                          {'bulk', knm_number:knm_numbers()} |
+-spec find_numbers(ne_binary(), pos_integer(), knm_search:options()) ->
+                          {'ok', list()} |
```
Comment (jamesaimonetti) [inline core/kazoo_number_manager/src/carriers/knm_other.erl]: From a specific type to a generic list?

File: core/kazoo_number_manager/src/knm_search.erl [hunk 2]
```diff
@@ -299,6 +299,10 @@ wait_for_search(N, Options) ->
             lager:debug("~s found numbers", [_Carrier]),
             gen_listener:cast(?MODULE, {'add_result', Numbers}),
             wait_for_search(N - 1, Options);
+        {_Carrier, {bulk, Numbers}} ->
```
Comment (jamesaimonetti) [inline core/kazoo_number_manager/src/knm_search.erl]: I miss tickies

File: core/kazoo_number_manager/test/knm_telnyx_test.erl [hunk 3]
```diff
@@ -31,12 +36,15 @@ find_numbers(Options) ->
      || {Prefix, Limit} <- [{<<"301359">>, 5}
                            ,{<<"800">>, 2}
                            ],
-        Results <- [knm_carriers:find(Prefix, [{'quantity',Limit}|Options])]
+        Options <- [[{quantity, Limit}
```
Comment (jamesaimonetti) [inline core/kazoo_number_manager/test/knm_telnyx_test.erl]: Why a list of a list of options?

**Comment by jamesaimonetti** [comment]
[Test failures](https://gist.github.com/jamesaimonetti/c37c0c88d52b7fa6b9a168631d4ba73d)

File: core/kazoo_number_manager/test/knm_telnyx_test.erl (hunk 3 above)
Comment (fenollp) [inline core/kazoo_number_manager/test/knm_telnyx_test.erl]: Run the code in your head

**Comment by fenollp** [comment]
Tests pass in CI and locally. Please revert your changes

File: core/kazoo_number_manager/src/carriers/knm_other.erl (hunk 1 above)
Comment (jamesaimonetti) [inline core/kazoo_number_manager/src/carriers/knm_other.erl]: Still wondering why you changed these to more generic versions. It breaks the behaviour's expected return:

```
knm_other.erl:60: The return type {'bulk',[any()]} | {'ok',[any()]} in the specification of find_numbers/3 is not a subtype of {'bulk',[knm_number:knm_number()]} | {'error',_} | {'ok',[knm_number:knm_number()]}, which is the expected return type for the callback of knm_gen_carrier behaviour
```

**Pull request merged**